phishingpage.py

```diff
@@ -0,0 +1,153 @@
+"""
+This module was made to handle all the phishing related operations for
+Wifiphisher.py
+"""
+
+import urllib
+import os
+
+# set of dictionaries to store URLs and names
+LINKSYS = {"index.html": "http://pastebin.com/raw.php?i=b0Uz1sta",
+           "Linksys_logo.png": "https://i.imgur.com/slBTPcu.png"}
+TEMPLATE_TO_URL = {"linksys": LINKSYS, "minimal": None,
+                   "connection_reset": None, "office365": None}
+
+PHISHING_PAGES_DIR = "phishing-pages/"
+
+
+class UrlNotAvailable(Exception):
+    """ Exception class to raise in case of a invalid URL. """
+    def __init__(self):
+        Exception.__init__(self, "The given URL is not available!")
+
+
+class TemplateNotAvailable(Exception):
+    """ Exception class to raise in case of non-existence template. """
+    def __init__(self):
+        Exception.__init__(self, "The given template is not available!")
+
+
+class ArgumentIsNotAString(Exception):
+    """ Exception class to raise in case of non-string input. """
+    def __init__(self):
+        Exception.__init__(self, "The given argument is not a string!")
+
+
+def grab_online(template):
+    """
+    Make a new folder with the name provided in template and fetch all the
+    files for the given template from the internet.
+
+    Args:
+        template (str): The name of requested template.
+
+    Returns:
+        None
+
+    Raises:
+        TemplateNotAvailble: If the given template is not in the database.
+        ArgumentIsNotAString: if the given template is not a string.
+    """
+    # check to see if template is a string
+    if not type(template) is str:
+        raise ArgumentIsNotAString()
+
+    # check if user's choice exists in dictionary
+    if template in TEMPLATE_TO_URL:
+
+        # make a new folder
+        os.makedirs(get_path(template))
+
+        # loop through template dictionary
+        for name in TEMPLATE_TO_URL[template]:
+
+            # get the URL
+            url = TEMPLATE_TO_URL[template][name]
+
+            # check if URL's exist
+            if url_check(url):
+                # download the files
+                urllib.urlretrieve(url, (get_path(template) + "/" + name))
+
+    else:
+
+        raise TemplateNotAvailable()
+
+
+def exists(dir_path):
```

Comment:
Also note here that if you find a incomplete download (some of the required files or the empty folder) you should cleanup and download from scratch again.